emmaa_service/api.py

```diff
@@ -1502,33 +1502,26 @@ def get_statement_evidence_page():
                 'path_stmt_counts', [])
             stmt_counts_dict += Counter(dict(stmt_counts))
         all_stmts = _load_stmts_from_cache(model, date)
-        for stmt in all_stmts:
-            for stmt_hash in stmt_hashes:
-                if str(stmt.get_hash(refresh=True)) == str(stmt_hash):
-                    stmts.append(stmt)
+        stmts = [stmt for stmt in all_stmts
+                 if str(stmt.get_hash(refresh=True)) in stmt_hashes]
```

Comment:
It looks like stmt_hashes is still a list here and below - I think we could get mileage out of making it a set here and below where it's used for checking membership since doing so in a long list is a lot slower.

Reply:
good point, changed it to set